examples/rvc-app/rvc-common/src/rvc-service-area-delegate.cpp

```diff
@@ -0,0 +1,467 @@
+    {
+        // not sorting list, number of locations normally expected to be small, max 255
+        mSupportedLocations.push_back(newLocation);
+        listIndex = (uint32_t) mSupportedMaps.size() - 1; // new element is last in list
+        ret_value = true;
+    }
+    else
+    {
+        ChipLogError(Zcl, "AddSupportedLocation %u - supported locations list is already at maximum size %u",
+                     newLocation.locationID, (uint32_t) kMaxNumSupportedLocations);
+    }
+
+    return ret_value;
+}
+
+bool RvcServiceAreaDelegate::ModifySupportedLocation(uint32_t listIndex, const LocationStructureWrapper & modifiedLocation)
+{
+    bool ret_value = false;
+
+    // The server instance (caller) is responsible for ensuring that there are no duplicate location IDs, list size not exceeded,
+    // etc.
+
+    // Double-check that locationID's match.
+    VerifyOrExit((modifiedLocation.locationID == mSupportedLocations[listIndex].locationID),
+                 ChipLogError(Zcl, "ModifySupportedLocation - new locationID %u does not match existing locationID %u",
+                              modifiedLocation.locationID, mSupportedLocations[listIndex].locationID));
+
+    // checks passed, update the attribute
+    mSupportedLocations[listIndex] = modifiedLocation;
+    ret_value                      = true;
+
+exit:
+
+    return ret_value;
+}
+
+bool RvcServiceAreaDelegate::ClearSupportedLocations()
+{
+    bool ret_value = false;
+
+    if ((uint32_t) mSupportedLocations.size() > 0)
+    {
+        mSupportedLocations.clear();
+        ret_value = true;
+    }
+
+    return ret_value;
+}
+
+//*************************************************************************
+// Supported Maps accessors
+
+bool RvcServiceAreaDelegate::IsSupportedMapChangeAllowed()
+{
+    return true; // TODO
+}
+
+uint32_t RvcServiceAreaDelegate::GetNumberOfSupportedMaps()
+{
+    return (uint32_t) mSupportedMaps.size();
+}
+
+bool RvcServiceAreaDelegate::GetSupportedMapByIndex(uint32_t listIndex, MapStructureWrapper & aSupportedMap)
+{
+    bool ret_value = false;
+
+    if (listIndex < (uint32_t) mSupportedMaps.size())
+    {
+        aSupportedMap = mSupportedMaps[listIndex];
+        ret_value     = true;
+    }
+
+    return ret_value;
+};
+
+bool RvcServiceAreaDelegate::GetSupportedMapById(uint8_t aMapId, uint32_t & listIndex, MapStructureWrapper & aSupportedMap)
+{
+    // We do not need to reimplement this method as it's already done by the SDK.
+    // We are reimplementing this method, still using linear search, but with some optimization on the SDK implementation
+    // since we have direct access to the list.
+    bool ret_value = false;
+
+    listIndex         = 0;
+    uint32_t listSize = (uint32_t) mSupportedMaps.size();
+
+    while (listIndex < listSize)
+    {
+        if (mSupportedMaps[listIndex].mapID == aMapId)
+        {
+            aSupportedMap = mSupportedMaps[listIndex];
+            ret_value     = true;
+            break;
```

Comment:
```suggestion
            aSupportedMap = mSupportedMaps[listIndex];
            return true;
```

please switch to early return if possible throughout the code - the pattern of goto-exit and single variable for return makes the code paths harder to follow (one has to see hte original assignment and validate how all code paths interact with the return value) - code is longer and harder to review and maintain (and as soon as we add a goto exit, it encourages all variables to be declared at the top which goes against minimizing scope of variables)

Reply:
This has been updated [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/63a7ae82ac9c287303d0e009cf403c55b1fe4caa).